test/test_clean_element.rb

```diff
@@ -0,0 +1,190 @@
+      :default    => 'foo',
+      :restricted => 'foo',
+      :basic      => '<a rel="nofollow">foo</a>',
+      :relaxed    => '<a>foo</a>'
+    },
+
+    'protocol-based JS injection: long UTF-8 encoding without semicolons' => {
+      :html       => '<a href=&#0000106&#0000097&#0000118&#0000097&#0000115&#0000099&#0000114&#0000105&#0000112&#0000116&#0000058&#0000097&#0000108&#0000101&#0000114&#0000116&#0000040&#0000039&#0000088&#0000083&#0000083&#0000039&#0000041>foo</a>',
+      :default    => 'foo',
+      :restricted => 'foo',
+      :basic      => '<a rel="nofollow">foo</a>',
+      :relaxed    => '<a>foo</a>'
+    },
+
+    'protocol-based JS injection: hex encoding' => {
+      :html       => '<a href="javascript&#x3A;">foo</a>',
+      :default    => 'foo',
+      :restricted => 'foo',
+      :basic      => '<a rel="nofollow">foo</a>',
+      :relaxed    => '<a>foo</a>'
+    },
+
+    'protocol-based JS injection: long hex encoding' => {
+      :html       => '<a href="javascript&#x003A;">foo</a>',
+      :default    => 'foo',
+      :restricted => 'foo',
+      :basic      => '<a rel="nofollow">foo</a>',
+      :relaxed    => '<a>foo</a>'
+    },
+
+    'protocol-based JS injection: hex encoding without semicolons' => {
+      :html       => '<a href=&#x6A&#x61&#x76&#x61&#x73&#x63&#x72&#x69&#x70&#x74&#x3A&#x61&#x6C&#x65&#x72&#x74&#x28&#x27&#x58&#x53&#x53&#x27&#x29>foo</a>',
+      :default    => 'foo',
+      :restricted => 'foo',
+      :basic      => '<a rel="nofollow">foo</a>',
+      :relaxed    => '<a>foo</a>'
+    },
+
+    'protocol-based JS injection: null char' => {
+      :html       => "<img src=java\0script:alert(\"XSS\")>",
+      :default    => '',
+      :restricted => '',
+      :basic      => '',
+      :relaxed    => '<img>'
+    },
+
+    'protocol-based JS injection: invalid URL char' => {
+      :html       => '<img src=java\script:alert("XSS")>',
+      :default    => '',
+      :restricted => '',
+      :basic      => '',
+      :relaxed    => '<img>'
+    },
+
+    'protocol-based JS injection: spaces and entities' => {
+      :html       => '<img src=" &#14;  javascript:alert(\'XSS\');">',
+      :default    => '',
+      :restricted => '',
+      :basic      => '',
+      :relaxed    => '<img>'
+    },
+
+    'protocol whitespace' => {
+      :html       => '<a href=" http://example.com/"></a>',
+      :default    => '',
+      :restricted => '',
+      :basic      => '<a href="http://example.com/" rel="nofollow"></a>',
+      :relaxed    => '<a href="http://example.com/"></a>'
+    }
+  }
+
+  describe 'Basic config' do
+    before do
+      @s = Sanitize.new(Sanitize::Config::BASIC)
+    end
+
+    it 'should not choke on valueless attributes' do
+      @s.clean('foo <a href>foo</a> bar')
+        .must_equal 'foo <a href rel="nofollow">foo</a> bar'
+    end
+  end
+
+  describe 'Custom configs' do
+    it "should not allow relative URLs when relative URLs aren't whitelisted" do
```

Comment:
This test seems to be unrelated to the functionality added in this PR. Can you clarify why you feel it should be added?

Reply:
Just an artifact of my merging the tests from `master`. Will remove.